### ISSUE TITLE
`column_diffusion_timescale` does not work on GPU

### ISSUE THREAD

**Issue by jagoosw** [OceanBioME/OceanBioME.jl#142]
As noted in https://github.com/OceanBioME/OceanBioME.jl/pull/138, `column_diffusion_timescale` does not work on GPU.

This function seems unlikely to ever be needed on GPU since column models would probably never need the performance.

We could rethink how it works so please let us know if you need it!

**Comment by glwagner** [comment]
Can you link to the code? Otherwise, it's hard to follow.

**Comment by jagoosw** [comment]
Good idea: https://github.com/OceanBioME/OceanBioME.jl/blob/ac8419a1ce5a06a82be31588583636b212c09598/src/Utils/timestep.jl#L4-L17

**Comment by glwagner** [comment]
What does it do?

**Comment by jagoosw** [comment]
I don't recall all of the motivation for it but it was written to work out the diffusion timescale for "column" models with 1 dimension and a diffusivity given by a function. I think we wrote it originally because Oceananigans can't do it for diffusion functions?

**Comment by glwagner** [comment]
I don't think we have implemented automagical support for calculating the diffusion time-scale for a diffusion function --- though such support is not too difficult using `FunctionField` + field reductions.

But more saliently, we rarely have use for such a function because we use vertically-implicit time-stepping for vertical diffusion which is unconditionally stable. For horizontal diffusion, while it is theoretically possible to violate the diffusive CFL, typical horizontal diffusion models are implemented in an "LES" context, wherein the diffusive limitation is very similar to the advective limitation.

All of that said, it's usually better to implement something in Oceananigans if support is needed --- because if other people need it to, they might improve your implementation.

I think it's unlikely that this function belongs in the source code. It's specific to a certain closure setup and only tests the "first" closure within a tuple, because of the `[1]` in this line:

```julia
         @inbounds Δz2_ν[k] = Δzₖ^2 / model.closure.κ[1](0.0, 0.0, z[k], model.clock.time) # assumes all tracer closures are the same and x/y invariant 
```

**Comment by jagoosw** [comment]
Yeah, I agree. I recall someone was using it for reading in a diffusivity to replicate a mixed layer from observations or some analytical function, but it was quite a specific case and I could see a function field being a much easier solution.

**Comment by glwagner** [comment]
> Yeah, I agree. I recall someone was using it for reading in a diffusivity to replicate a mixed layer from observations or some analytical function, but it was quite a specific case and I could see a function field being a much easier solution.

Yes, that sounds like user code that accidentally ended up in the source code.

**Comment by johnryantaylor** [comment]
I think this was from Si's column model experiments where kappa is strongly dependent on time.  The issue isn't necessarily numerical stability (if using an implicit solver), but you still don't want the time step to be too large when diffusion is large for the purpose of accuracy.  Anyway, I agree that this is a very specific use case and doesn't need to be in the repo.

**Comment by glwagner** [comment]
It could make sense to add support within

https://github.com/CliMA/Oceananigans.jl/blob/main/src/TurbulenceClosures/turbulence_closure_diagnostics.jl

at some point.
